app/src/main/java/com/douglasbruce/tasky/core/data/database/dao/ReminderDao.kt

```diff
@@ -2,18 +2,26 @@ package com.douglasbruce.tasky.core.data.database.dao
 
 import androidx.room.Dao
 import androidx.room.Query
+import androidx.room.Transaction
+import androidx.room.Upsert
 import com.douglasbruce.tasky.core.data.database.model.ReminderEntity
 import kotlinx.coroutines.flow.Flow
 
 @Dao
 interface ReminderDao {
+    @Upsert
+    suspend fun upsertReminder(reminder: ReminderEntity)
+
+    @Transaction
```

Comment:
This would also just need the `Upsert` annotation. Room accepts both lists and single items